nibabel/openers.py

```diff
@@ -145,3 +145,13 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close_if_mine()
+
+    @classmethod
+    def register_extension(opener_klass, ext, func):
+        """Decorator"""
+        def decorate(klass):
+            assert ext not in opener_klass.compress_ext_map, \
+                "Cannot redefine extension-function mappings."
+            opener_klass.compress_ext_map[ext] = func
```

Comment:
This will modify the Opener class globally no?   So all 'Opener' instances will now effectively become BinOpener instances?